plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -256,534 +254,666 @@ struct block_time_tracker {
+      _unapplied_transactions.add_aborted(chain.abort_block());
+      _idle_trx_time = fc::time_point::now();
+   }
 
-      bool on_incoming_block(const signed_block_ptr& block, const std::optional<block_id_type>& block_id, const block_state_ptr& bsp) {
-         auto& chain = chain_plug->chain();
-         if ( _pending_block_mode == pending_block_mode::producing ) {
-            fc_wlog( _log, "dropped incoming block #${num} id: ${id}",
-                     ("num", block->block_num())("id", block_id ? (*block_id).str() : "UNKNOWN") );
-            return false;
-         }
+   bool on_incoming_block(const signed_block_ptr& block, const std::optional<block_id_type>& block_id, const block_state_ptr& bsp) {
+      auto& chain = chain_plug->chain();
+      if (_pending_block_mode == pending_block_mode::producing) {
+         fc_wlog(
+            _log, "dropped incoming block #${num} id: ${id}", ("num", block->block_num())("id", block_id ? (*block_id).str() : "UNKNOWN"));
+         return false;
+      }
 
-         // start a new speculative block, speculative start_block may have been interrupted
-         auto ensure = fc::make_scoped_exit([this](){
-            schedule_production_loop();
-         });
+      // start a new speculative block, speculative start_block may have been interrupted
+      auto ensure = fc::make_scoped_exit([this]() { schedule_production_loop(); });
 
-         const auto& id = block_id ? *block_id : block->calculate_id();
-         auto blk_num = block->block_num();
+      const auto& id      = block_id ? *block_id : block->calculate_id();
+      auto        blk_num = block->block_num();
 
-         auto now = fc::time_point::now();
-         if (now - block->timestamp < fc::minutes(5) || (blk_num % 1000 == 0)) // only log every 1000 during sync
-            fc_dlog(_log, "received incoming block ${n} ${id}", ("n", blk_num)("id", id));
+      auto now = fc::time_point::now();
+      if (now - block->timestamp < fc::minutes(5) || (blk_num % 1000 == 0)) // only log every 1000 during sync
+         fc_dlog(_log, "received incoming block ${n} ${id}", ("n", blk_num)("id", id));
 
-         EOS_ASSERT( block->timestamp < (now + fc::seconds( 7 )), block_from_the_future,
-                     "received a block from the future, ignoring it: ${id}", ("id", id) );
+      EOS_ASSERT(block->timestamp < (now + fc::seconds(7)),
+                 block_from_the_future,
+                 "received a block from the future, ignoring it: ${id}",
+                 ("id", id));
 
-         /* de-dupe here... no point in aborting block if we already know the block */
-         auto existing = chain.fetch_block_by_id( id );
-         if( existing ) { return true; } // return true because the block is valid
+      /* de-dupe here... no point in aborting block if we already know the block */
+      auto existing = chain.fetch_block_by_id(id);
+      if (existing) {
+         return true;
+      } // return true because the block is valid
```

Comment:
Move comment now that no longer one line.

Reply:
done